scripts/leetcode_directory_md.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+from urllib.request import Request, urlopen
+from pathlib import Path
+from os import listdir
+import json
+from dataclasses import dataclass
+
+
+@dataclass
+class Task:
+    id: str
+    title: str
+    solution: str
+    difficulty: str
+
+def fetch_leetcode_folder_tasks(solutions_folder: Path) -> list[Task]:
+    # Fetch tasks info from the leetcode API.
+    req = Request(
+        url="https://leetcode.com/api/problems/algorithms/", 
+        headers={"User-Agent": "Mozilla/5.0 (X11; U; Linux i686) Gecko/20071127 Firefox/2.0.0.11"}
+    )
+
+    content = urlopen(req).read()
+    content_dict = json.loads(content)
+
+    raw_tasks_id_dict = {}
+
+    for task in content_dict["stat_status_pairs"]:
+        task_stat = task["stat"]
+        raw_tasks_id_dict[str(task_stat["frontend_question_id"])] = task
+
+    # Generate result tasks info to be inserted into the document
+    tasks_list = []
+
+    difficulty={
+        1: 'Easy',
+        2: 'Medium',
+        3: 'Hard'
+    }

```

Comment:
Please use consistent formatting - you sometimes use `'` and sometimes use `"` when defining strings.